builder/builder-module.c

```diff
@@ -1343,9 +1372,10 @@ builder_module_build (BuilderModule  *self,
         }
     }
 
-  has_configure = g_file_query_exists (configure_file, NULL);
+  if (!simple)
+    has_configure = g_file_query_exists (configure_file, NULL);
 
-  if (!has_configure && !self->no_autogen)
+  if (!simple && !has_configure && !self->no_autogen)
```

Comment:
We can, I didn't do it originally because it makes a pretty big diff with mostly whitespace changes.

However it also splits the code in a weird way:

```c
  if (!simple)
    {
      // Do lots of preparation work, like finding the configure files, the make command, etc.
      ...
      // Eventually, run the build
      if (!build (app_dir, self->name, context, source_dir, build_dir_relative, build_args, env, error,
                  make_cmd, make_j ? make_j : skip_arg, make_l ? make_l : skip_arg, strv_arg, self->make_args, NULL))
        return FALSE;
    }

  if (self->build_commands)
    {
      for (i = 0; self->build_commands[i] != NULL; i++)
        {
          g_print ("%s\n", self->build_commands[i]);
          if (!build (app_dir, self->name, context, source_dir, build_dir_relative, build_args, env, error,
                      "/bin/sh", "-c", self->build_commands[i], NULL))
            return FALSE;
        }
    }

  // Then back to !simple in order to run the install command
  if (!simple)
    {
      if (!build (app_dir, self->name, context, source_dir, build_dir_relative, build_args, env, error,
                  make_cmd, "install", strv_arg, self->make_install_args, NULL))
        return FALSE;
    }
```

Maybe the trick would be to add the `make`/`make install` at the beginning/end of the `self->build_commands` list, and then just always run this list?

That is, all build systems would end up being special cases of the "simple" one, rather than the "simple" buildsystem being a special case with only "build-commands".